prql-compiler/src/sql/gen_expr.rs

```diff
@@ -184,6 +183,60 @@ pub(super) fn translate_literal(l: Literal, ctx: &Context) -> Result<sql_ast::Ex
     })
 }
 
+fn translate_datetime_literal(
+    data_type: sql_ast::DataType,
+    value: String,
+    ctx: &Context,
+) -> sql_ast::Expr {
+    if ctx.dialect.is::<crate::sql::dialect::SQLiteDialect>() {
```

Comment:
@aljazerzen let us know if you have thoughts on this vs methods on `DialectHandler`. This approach seems good...

Reply:
IMO, more or less we will need a similar way to tell which target we are generating the query to. This would be more helpful if we are going to generate query for languages beyond SQL-like.

Or maybe another approach is we push down the full generation into each dialect, and for most of them, we can share the same implementation as we currently doing. It's like the folder pattern we are using in the parse phase.

But I do feel like this needs more discussions / designs. I used this trick as 1) I couldn't come up a good function name for this datetime settings; 2) this datetime usage seems limited to sqlite only, therefore, I feel it's better to explicitly to call it out.